frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1589,14 +1643,21 @@ impl<T: Config> ElectionProvider for Pallet<T> {
 	fn elect() -> Result<BoundedSupportsOf<Self>, Self::Error> {
 		match Self::do_elect() {
 			Ok(supports) => {
-				// All went okay, record the weight, put sign to be Off, clean snapshot, etc.
+				// All went okay, record the weight, put sign to be Off, clean snapshot, set the
+				// block of the last successful election, etc.
 				Self::weigh_supports(&supports);
 				Self::rotate_round();
+
 				Ok(supports)
 			},
 			Err(why) => {
-				log!(error, "Entering emergency mode: {:?}", why);
-				Self::phase_transition(Phase::Emergency);
+				// Enters in emergency phase only if election failed and the number of signed and
+				// unsigned blocks since last round rotate is higher than
+				// `T::MinBlocksBeforeEmergency`.
+				if !Self::emergency_phase_throttling() {
```

Comment:
Not elegant to re-check/re-execute `emergency_phase_throttling` here -- One error variatnt should be `Err(Throttling)` and then we match on it here.

Reply:
Or else, we can move the `phase_transition` to `do_elect`? 

either way, calling `emergency_phase_throttling` is a bit icky to me.

Reply:
I ended up with wrapping the election error with an `ElectionError::Throttling(Box<ElectionError>)`. Here we can check  if the error is wrapped in the throttling variant, while main election error can be returned by deferring the inner value.